opp_path_selection/include/opp_path_selection/filter_impl.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OPP_PATH_SELECTION_FILTER_IMPL_H
+#define OPP_PATH_SELECTION_FILTER_IMPL_H
+
+#include "opp_path_selection/filter.h"
+
+#include <pcl/filters/extract_indices.h>
+#include <pcl/sample_consensus/ransac.h>
+#include <pcl/sample_consensus/sac_model_plane.h>
+#include <pcl/segmentation/sac_segmentation.h>
+
+#include <ros/ros.h>
+
+namespace opp_path_selection
+{
+namespace data_filtering
+{
+template <typename PointT>
+bool planeFit(const CloudPtr<PointT> input_cloud,
+              Cloud<PointT>& output_cloud,
+              pcl::ModelCoefficients::Ptr plane_coefficients,
+              const double threshold)
```

Comment:
I don't think there is a need to have the implementation of this function in a separate header than its definition. The value of separating the definition and implementation comes when you create a library that explicitly instantiates this template for some types so that users of this function can link against that library rather than compiling this function. Since we're not explicitly instantiating this function for any `PointT`, I would suggest combining this with its definition

Reply:
This code mirrors the existing opp_area_selection and duplicates much of that code and structures. If one is changed then both should be to match your suggestions. Let's leave that for a TODO in another PR.